tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -129,7 +150,19 @@ def index_impl(self):
 
   def scalars_impl(self, tag, run, experiment, output_format):
     """Result of the form `(body, mime_type)`."""
-    if self._db_connection_provider:
+    if self._data_provider:
+      all_scalars = self._data_provider.read_scalars(
+          experiment_id=None,  # experiment support not yet implemented
+          owner_plugin=metadata.PLUGIN_NAME,
+          downsample_to=1000,
```

Comment:
Perhaps factor out the constant so we can explain where it comes from?

Reply:
Sure.